server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -147,33 +162,96 @@ private Setting<?> createSetting(
                     ? Setting.boolSetting(key, (boolean) defaultValue, propertyArray)
```

Comment:
We don't use the `BooleanParser` here.  That's okay because `boolSetting()` uses the `key` and `propertyArray` to recreate it.  

However we still use the `BooleanParser` in logic elsewhere in the code, creating dead code that will never execute, for example the `writeParser` switch/case for boolean (which means it will never be read by the `StreamInput` constructor either.

So I'd suggest either creating the (Writeable) `BooleanParser` here so the other code is actually used, or removing the `Boolean` case from the reading and writing switches later in this code and deleting the entire `BooleanParser` code.

Reply:
ACK

Reply:
Since we are not using the BooleanParser, I think we can delete it.